### PR TITLE
doc: fix documentation for the Go library

### PR DESCRIPTION
**Pull request by Wwwsylvia**
1. Clean up the old content
2. Update the client library overview
3. Add introduction and quick start for the Go library page

TODOs:
1. Move the quick start examples to the oras-go repo
2. Add architecture diagram for oras-go

Related to #67, https://github.com/oras-project/oras-go/issues/341
Fixes https://github.com/oras-project/oras-go/issues/421
Fixes https://github.com/oras-project/oras-go/issues/424
Fixes https://github.com/oras-project/oras-go/issues/430

### CODE REVIEW

File: docs/Client_Libraries/0_go.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	if err != nil {
+		panic(err)
+	}
+	repo.Client = &auth.Client{
+		Cache: auth.DefaultCache,
+		Credential: auth.StaticCredential(reg, auth.Credential{
+			Username: "username",
+			Password: "password",
+		}),
+	}
```
Comment (shizhMSFT) [inline docs/Client_Libraries/0_go.md:154]: nit: you may mention this block of code is not required for anonymous pull.

File: docs/Client_Libraries/0_go.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	defer fs.Close()
+
+	// 1. Connect to a remote repository
+	ctx := context.Background()
+	reg := "myregistry.example.com"
+	repo, err := remote.NewRepository(reg + "/myrepo")
+	if err != nil {
+		panic(err)
+	}
+	repo.Client = &auth.Client{
```
Comment (shizhMSFT) [inline docs/Client_Libraries/0_go.md:148]: Please note about the retry client.

File: docs/Client_Libraries/0_go.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+	// 2. Copy from the remote repository to the file store
+	tag := "latest"
+	manifestDescriptor, err := oras.Copy(ctx, repo, tag, fs, tag, oras.DefaultCopyOptions)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println("manifest descriptor:", manifestDescriptor)
+}
+
```
Comment (shizhMSFT) [inline docs/Client_Libraries/0_go.md:164]: nit: remove empty line

File: docs/Client_Libraries/0_go.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+		}),
+	}
+
+	// 3. Copy from the file store to the remote repository
+	_, err = oras.Copy(ctx, fs, tag, repo, tag, oras.DefaultCopyOptions)
+	if err != nil {
+		panic(err)
+	}
+}
+
```
Comment (shizhMSFT) [inline docs/Client_Libraries/0_go.md:115]: nit: remove empty line

File: docs/Client_Libraries/0_go.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
+# Go
+
+Source code: [github.com/oras-project/oras-go](https://github.com/oras-project/oras-go)
+
+## Introduction
+
+The ORAS Go client library provides the ability to replicate artifacts between different [Targets](../#target).  
+Furthermore, the version `v2` is a registry client conforming [image-spec v1.1.0-rc.2](https://github.com/opencontainers/image-spec/releases/tag/v1.1.0-rc2) and [distribution-spec v1.1.0-rc1](https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md).
+
+Using the ORAS Go client library, you can develop your own push/pull experience:
```
Comment (FeynmanZhou) [inline docs/Client_Libraries/0_go.md:10]: Suggest changing `your own push/pull experience` to `your own registry client` since push/pull is narrow.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: docs/Client_Libraries/0_go.md (hunk 5 above)
Comment (Wwwsylvia) [reply docs/Client_Libraries/0_go.md]: Updated

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Pull request merged**